kolibri/core/assets/test/core_app_mediator.js

```
@@ -1,499 +1,517 @@
+      });
+      it('should call the _registerOneTimeEvents method', () => {
+        assert(this._registerOneTimeEvents.called);
+      });
+      it('should pass the kolibriModule to the _registerOneTimeEvents method', () => {
+        assert(this._registerOneTimeEvents.calledWith(this.kolibriModule));
+      });
+      it('should call the emit method', () => {
+        assert(this.emit.called);
+      });
+      it('should pass the kolibriModule to the emit method', () => {
+        assert(this.emit.calledWith('kolibri_register', this.kolibriModule));
+      });
+      it('should call the _executeCallbackBuffer method', () => {
+        assert(this._executeCallbackBuffer.called);
+      });
+      it('should call pass the kolibriModule to the _executeCallbackBuffer method', () => {
+        assert(this._executeCallbackBuffer.calledWith(this.kolibriModule));
+      });
+      it('should put the kolibriModule into the kolibriModule registry', () => {
+        assert.deepEqual(
+          this.mediator._kolibriModuleRegistry[this.kolibriModule.name], this.kolibriModule
+        );
+      });
     });
-    describe('_registerEvents method', function() {
-        it('should not throw a TypeError due to incorrect assignment of this when the eventListenerMethod is called', function() {
-            var self = this;
-            assert.doesNotThrow(function () {
-                self.mediator._registerEvents({
-                    name: 'test',
-                    events: {
-                        event: 'method'
-                    }
-                }, 'events', self.mediator._registerRepeatedEventListener);
-            }, TypeError);
-        });
+    describe('called with invalid input', () => {
+      beforeEach(() => {
+        this.kolibriModule = undefined;
+        try {
+          this.mediator.registerKolibriModuleSync(this.kolibriModule);
+        } catch (e) {} // eslint-disable-line no-empty
```

should this be an empty catch?